arangod/RestHandler/RestSimpleQueryHandler.cpp

```diff
@@ -229,3 +236,94 @@ void RestSimpleQueryHandler::allDocumentKeys() {
   // now run the actual query and handle the result
   processQuery(s);
 }
+
+static void buildExampleQuery(VPackBuilder& result,
+                              std::string const& cname,
+                              VPackSlice const& doc,
+                              size_t skip, size_t limit) {
+  TRI_ASSERT(doc.isObject());
+  std::string query = "FOR doc IN @@collection";
+  
+  result.add("bindVars", VPackValue(VPackValueType::Object));
+  result.add("@collection", VPackValue(cname));
+  size_t i = 0;
+  for (auto pair : VPackObjectIterator(doc, true)) {
+    std::string key = basics::StringUtils::replace(pair.key.copyString(), "`", "");
+    key = basics::StringUtils::join(basics::StringUtils::split(key, "."), "`.`");
+    std::string istr = std::to_string(i++);
+    query.append(" FILTER doc.`").append(key).append("` == @value").append(istr);
+    result.add(std::string("value") + istr, pair.value);
+  }
+  result.close();
+  
+  if (limit > 0 || skip > 0) {
+    query.append(" LIMIT ").append(std::to_string(skip))
+         .append(", ").append(limit > 0 ? std::to_string(limit) : "null")
+         .append(" ");
+  }
+  query.append(" RETURN doc");
+  
+  result.add("query", VPackValue(query));
+}
+
+//////////////////////////////////////////////////////////////////////////////
+/// @brief return a cursor with all documents matching the example
+//////////////////////////////////////////////////////////////////////////////
+
+void RestSimpleQueryHandler::byExample() {
+  bool parseSuccess = true;
+  std::shared_ptr<VPackBuilder> parsedBody =
+  parseVelocyPackBody(parseSuccess);
+  if (!parseSuccess) {
+    // error message generated in parseVelocyPackBody
+    return;
+  }
+  VPackSlice body = parsedBody.get()->slice();
+  if (!body.isObject() || !body.hasKey("example") ||
+      !body.get("example").isObject()) {
+    generateError(ResponseCode::BAD, TRI_ERROR_BAD_PARAMETER);
+    return;
+  }
+  
+  size_t skip = basics::VelocyPackHelper::getNumericValue(body, "skip", 0);
```

Comment:
not sure what happens here if the number passed by end user is negative...
will it be handled properly?